starter_code/src/movies.js

```diff
@@ -1,15 +1,166 @@
 /* eslint no-restricted-globals: 'off' */
 
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+//..sort by year
+function orderByYear(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array))
+    return orderByTitle(array).sort((a, b) => a["year"] - b["year"]);
+}
+
+//..sort by title (only if same year)
+function orderByTitle(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array))
+    return newMoviesArray.sort(function(a, b) {
+      if (a["title"] < b["title"]) {
+        return -1;
+      } 
+      if (a["title"] > b["title"]) {
+        return 1;
+      }
+      if (a["title"] == b["title"]) {
+        return 0;
+      }
+    }); 
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function theSpielbergMovies(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array));
+    const spielbergMovies = newMoviesArray.filter(theDirector => theDirector["director"] == "Steven Spielberg");
+    return spielbergMovies;
+}
+  
+function theDramaMovies(array) {
+    const newMoviesArray = JSON.parse(JSON.stringify(array));
+    const theDramaMovies = newMoviesArray.filter(theDrama => theDrama["genre"].includes("Drama"));
+    return theDramaMovies;
+}
+
+function howManyMovies(array) {
+    const dramaArray = theDramaMovies(array);
+    return theSpielbergMovies(dramaArray).length;
+}
 
```

Comment:
Another possible way to solve this would be to do both at the same time. Like this  
function howManyMovies(someMovies) {
  return someMovies.filter(
    eachMovie =>
      eachMovie.director === 'Steven Spielberg' &&
      eachMovie.genre.includes('Drama')
  ).length;
}